library/manageiq.py

```diff
@@ -189,16 +205,17 @@ def add_or_update_provider(self, provider_name, provider_type, endpoints):
             and a short message describing the operation executed
         """
         message = ""
+        zone_id = self.find_zone_by_name(zone or 'default')
```

Comment:
What if not found and we get zone_id `None`?
- Does update_provider deal with that?
- Does add_new_provider deal with that?

Or do all installations have a `default` zone?